src/software/world/field.h

```diff
@@ -155,6 +155,13 @@ class Field
      */
     Rectangle friendlyDefenseArea() const;
 
+    /**
+     * Gets the friendly defense area extended to field boundary as a Rectangle.
+     *
+     * @return defense area of the friendly team
+     */
+    Rectangle friendlyDefenseAreaToBoundary() const;
```

Comment:
Javadoc doesn't specify it's only expanded to the boundary in the y direction. Based on the name I'd expect it to expand in all directions (or at least +y, -y, -x).

Reply:
How about change to `friendlyDefenseAreaToYBoundary`?

Reply:
Wait I just realized, why exactly do we need this function? I see it's used for the `FRIENDLY_DEFENSE_AREA` motion constraint, but I don't understand why it needs to be expanded to the boundary. Wouldn't it just be the existing defense area?

Reply:
yes but to resolve #980 we want it expanded to the boundaries

Reply:
changed the interface so removed this function completely

Reply:
#980 was for the friendly and enemy halfs, not the defense area. Don't think we saw issues with the defense area?